qiskit/qobj/converters/pulse_instruction.py

```diff
@@ -308,9 +308,9 @@ def parsed_fun(*args, **kwargs):
         subs = {}
         matched_params = []
         if args:
-            subs.update({symbols[i]: arg for i, arg in enumerate(args)})
+            subs.update({Symbol(symbols[i]): arg for i, arg in enumerate(args)})
             matched_params += list(params[i] for i in range(len(args)))
-        elif kwargs:
+        if kwargs:
             subs.update({local_dict[key]: value for key, value in kwargs.items()
                          if key in local_dict})
             matched_params += list(key for key in kwargs if key in params)
```

Comment:
Perhaps you could raise an error (`QiskitError`) if the supplied `subs` do not cover the required params and list the missing substitutions? Please remember to update the `Raises` docstring.